SwiftReflector/SwiftXmlReflection/BaseDeclaration.cs

```diff
@@ -112,6 +112,44 @@ static ProtocolDeclaration ThisOrParentProtocol (BaseDeclaration self)
 			return null;
 		}
 
+		public bool IsProtocolWithAssociatedTypesFullPath (NamedTypeSpec named, TypeMapper typeMap)
+		{
+			return OwningProtocolFromGenericWithFullPath (named, typeMap) != null;
+		}
+
+		public ProtocolDeclaration OwningProtocolFromGenericWithFullPath (NamedTypeSpec named, TypeMapper typeMap)
+		{
+			if (named.Name.Contains (".")) {
+				var parts = named.Name.Split ('.');
```

Comment:
why both `Contains` and `Split` ?
if you're not worried about perf you could do `Split` only (if more than one part then `.` was present)
if you think it's an hot path then an `IndexOf` and a `Substring`) looks better: no array allocation and you're only concerned with the first part